frontend/src/components/mine/Applications/MineApplicationTable.js

```diff
@@ -0,0 +1,137 @@
+import React from "react";
+import { Table, Button } from "antd";
+import NullScreen from "@/components/common/NullScreen";
+import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
+import * as Strings from "@/constants/strings";
+import * as Permission from "@/constants/permissions";
+import { getDropdownApplicationStatusOptions } from "@/selectors/staticContentSelectors";
+import CustomPropTypes from "@/customPropTypes";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { EDIT_OUTLINE } from "@/constants/assets";
+
+/**
+ * @class  MineApplicationTable - displays a table of applicationsfor a mine.
+ */
+
+const propTypes = {
+  applications: PropTypes.arrayOf(CustomPropTypes.application),
+  applicationStatusOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+  isMajorMine: PropTypes.bool.isRequired,
+  openEditApplicationModal: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  applications: [],
+};
+
+const columns = [
+  {
+    title: "Application No.",
+    dataIndex: "applicationNo",
+    key: "applicationNo",
+    render: (text) => <div title="Application No.">{text}</div>,
+  },
+  {
+    title: "Status",
+    dataIndex: "status",
+    key: "status",
+    render: (text, record) => (
+      <div title="Status">
+        {record.applicationStatusOptions.find((item) => item.value === text).label}
```

Comment:
If this doesn't find a match, it will return `undefined`. `undefined.label` will throw a runtime error. We should add some safety around that

Reply:
I agree, at the moment however it is not a pattern we use and the cases in which this could happen are: 

1. For some reason the application status options have not been loaded. 
2. Somehow an application was added to our system with an invalid foreign key.

I can add a check in if you'd like, but we should probably take a look at all of our find() calls on the front end to see if this is a needed change.

Reply:
Changed.